sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1485,3 +1504,11 @@ def delete(self, name: str) -> None:
 
     def _is_materialized_view(self, model: Model) -> bool:
         return isinstance(model.kind, ViewKind) and model.kind.materialized
+
+
+def _intervals(snapshot: Snapshot, deployability_index: DeployabilityIndex) -> Intervals:
+    return (
+        snapshot.intervals
+        if deployability_index.is_deployable(snapshot)
+        else snapshot.dev_intervals
```

Comment:
are there any edge cases where since we've never run in dev, intervals is empty so we do the wrong thing?

Reply:
What would the wrong thing be? If it never ran in dev then we'd just recreate the dev table.